code/modules/station_goals/bluespace_tap.dm

```diff
@@ -0,0 +1,359 @@
+//Station goal stuff goes here
+/datum/station_goal/bluespace_tap
+	name = "Bluespace Tap"
+	var/goal = 500000
+
+/datum/station_goal/bluespace_tap/get_report()
+	return {"<b>Bluespace Mining Tap Experiment</b><br>
+	Our search for new resources to exploit is nearing a tremendous break-through. One of our research stations in a nearby sector has recently finished a prototype for a device called a Bluespace Tap.
+	It reaches through bluespace into other dimensions to shift through them for interesting objects.<br>
+	Due to unforseen circumstances the large-scale test of the prototype could not be completed on the original research station. It will instead be carried out on your station.
+	Acquire the circuit board, construct the device over a wire knot and feed it enough power to generate [goal] mining points by shift end.
+	<br><br>
+	Be advised that the device is experimental and might act in slightly unforseen ways if sufficiently powered.
+	<br>
+	Nanotrasen Science Directorate"}
+
+/datum/station_goal/bluespace_tap/on_report()
+	var/datum/supply_packs/misc/bluespace_tap/P = SSshuttle.supply_packs["[/datum/supply_packs/misc/bluespace_tap]"]
+	P.special_enabled = TRUE
+
+/datum/station_goal/bluespace_tap/check_completion()
+	if(..())
+		return TRUE
+	for(var/obj/machinery/power/bluespace_tap/T in GLOB.machines)
+		if(T.total_points >= goal)
+			to_chat(world, "<b>Bluespace Tap Highscore</b> : <span class='greenannounce'>[T.total_points]</span>")	
```

Comment:
Not sure this is the right place to do this, but it works out in practise without refactoring station goal completion code.